diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullableTypeRule.kt

```diff
@@ -0,0 +1,136 @@
+            "String" -> FixedParam(null, null, "", true)
+            else -> findFixableForCollectionParam(reference.text)
+        }
+    }
+
+    private fun findFixableForCollectionParam(referenceText: String): FixedParam? =
+            when(referenceText) {
+                "List", "Iterable" -> FixedParam(null, null, "emptyList()")
+                "Map" -> FixedParam(null, null, "emptyMap()")
+                "Array" -> FixedParam(null, null, "emptyArray()")
+                "Set" -> FixedParam(null, null, "emptySet()")
+                "Sequence" -> FixedParam(null, null, "emptySequence()")
+                "Queue" -> FixedParam(null, null, "LinkedList()")
+                "MutableList" -> FixedParam(null, null, "mutableListOf()")
+                "MutableMap" -> FixedParam(null, null, "mutableMapOf()")
+                "MutableSet" -> FixedParam(null, null, "mutableSetOf()")
+                "LinkedList" -> FixedParam(null, null, "LinkedList()")
+                "LinkedHashMap" -> FixedParam(null, null, "LinkedHashMap()")
+                "LinkedHashSet" -> FixedParam(null, null, "LinkedHashSet()")
+                else -> null
+            }
+
+    private fun findSubstitution(node: ASTNode, fixedParam: FixedParam) {
+        if (fixedParam.isString)
+            replaceValueForString(node)
+        else if (fixedParam.insertConstantType != null && fixedParam.insertType != null)
+            replaceValue(node, fixedParam.insertConstantType, fixedParam.insertType, fixedParam.textNode)
+        else
+            replaceValueByText(node, fixedParam.textNode)
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun replaceValueByText(node: ASTNode, nodeText: String) {
+        val newNode = KotlinParser().createNode(nodeText)
+        if (newNode.elementType == CALL_EXPRESSION) {
+            node.addChild(newNode, node.findChildByType(NULL))
+            node.removeChild(node.findChildByType(NULL)!!)
+        }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun replaceValue(node: ASTNode, insertConstantType: IElementType, insertType: IElementType, textNode: String) {
+        val value = CompositeElement(insertConstantType)
+        node.addChild(value, node.findChildByType(NULL)!!)
+        node.removeChild(node.findChildByType(NULL)!!)
+        value.addChild(LeafPsiElement(insertType, textNode))
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun replaceValueForString(node: ASTNode) {
+        val value = CompositeElement(STRING_TEMPLATE)
+        node.addChild(value, node.findChildByType(NULL)!!)
+        node.removeChild(node.findChildByType(NULL)!!)
+        value.addChild(LeafPsiElement(OPEN_QUOTE, ""))
+        value.addChild(LeafPsiElement(CLOSING_QUOTE, ""))
+    }
+
+    data class FixedParam(val insertConstantType: IElementType?, val insertType: IElementType?, val textNode: String, val isString: Boolean = false)
```

Comment:
```suggestion
    private data class FixedParam(val insertConstantType: IElementType?, val insertType: IElementType?, val textNode: String, val isString: Boolean = false)
```

Reply:
Also, why is second property called `insertConstantType`? As far as I can tell from code it is type of a composite element which will be inserted into code, isn't it?